Snowflake.Data/Core/SFDataConverter.cs

```diff
@@ -70,7 +70,8 @@ internal static object ConvertToCSharpVal(UTF8Buffer srcVal, SFDataType srcType,
                 }
                 else if (destType == typeof(Boolean))
                 {
-                    return srcVal.Buffer[0] == '1' || srcVal.Buffer[0] == 't' || srcVal.Buffer[0] == 'T';
+                    var val = srcVal.Buffer[srcVal.offset];

```

Comment:
You'll find that the unit tests will _fail_ if you remove this source code change. This is super critical as it can mutate data and return `false` for things that are actually `true`.